soccer/gameplay/skills/angle_receive.py

```diff
@@ -0,0 +1,259 @@
+
+    ## max difference between where we should be facing and where we are facing (in radians)
+    FaceAngleErrorThreshold = 8 * constants.DegreesToRadians
+
+    ## how much we're allowed to be off in the direction of the pass line
+    PositionYErrorThreshold = 0.06
+
+    ## how much we're allowed to be off side-to-side from the pass line
+    PositionXErrorThreshold = 0.03
+
+    DribbleSpeed = 70
+
+    ## we have to be going slower than this to be considered 'steady'
+    SteadyMaxVel = 0.04
+    SteadyMaxAngleVel = 3   # degrees / second
+
+    ## after this amount of time has elapsed after the kick and we haven't received the ball, we failed :(
+    ReceiveTimeout = 2
+
+    class State(enum.Enum):
+        ## we're aligning with the planned receive point
+        aligning = 1
+
+        ## being in this state signals that we're ready for the kicker to kick
+        aligned = 2
+
+        ## the ball's been kicked and we're adjusting based on where the ball's moving
+        receiving = 3
+
+    def __init__(self):
+        super().__init__()
+
+        self.ball_kicked = False
+        self._target_pos = None
+        self._ball_kick_time = 0
+
+        self._shot_time = 0
+        self._shot_occured = None
+        self._angle_facing = None
+
+
+        for state in AngleReceive.State:
+            self.add_state(state, behavior.Behavior.State.running)
+
+
+        self.add_transition(behavior.Behavior.State.start,
+                AngleReceive.State.aligning,
+                lambda: True,
+                'immediately')
+
+        self.add_transition(AngleReceive.State.aligning,
+                AngleReceive.State.aligned,
+                lambda: self.errors_below_thresholds() and self.is_steady() and not self.ball_kicked,
+                'steady and in position to receive')
+
+        self.add_transition(AngleReceive.State.aligned,
+                AngleReceive.State.aligning,
+                lambda: (not self.errors_below_thresholds() or not self.is_steady()) and not self.ball_kicked,
+                'not in receive position')
+
+        for state in [AngleReceive.State.aligning, AngleReceive.State.aligned]:
+            self.add_transition(state,
+                    AngleReceive.State.receiving,
+                    lambda: self.ball_kicked,
+                    'ball kicked')
+
+            self.add_transition(AngleReceive.State.receiving,
+                    behavior.Behavior.State.completed,
+                    lambda: self._shot_occured,
+                    'ball received!')
+
+            self.add_transition(AngleReceive.State.receiving,
+                    behavior.Behavior.State.failed,
+                    lambda: time.time() - self._ball_kick_time > AngleReceive.ReceiveTimeout,
+                    'ball missed :(')
+
+
+
+    ## set this to True to let the receiver know that the pass has started and the ball's in motion
+    # Default: False
+    @property
+    def ball_kicked(self):
```

Comment:
Take a look at `OurRobot.just_kicked` - I think it does the same thing as you're doing with this property, but is baked into the `OurRobot` class.